docs/root/api/starting_envoy.rst

```diff
@@ -184,6 +184,25 @@ This functionality is used for stat segmentation.
   // Swift
   builder.addVirtualClusters("[{\"name\":\"vcluster\",\"headers\":[{\"name\":\":path\",\"exact_match\":\"/v1/vcluster\"}]}]")
 
+~~~~~~~~~~~~~~~~~~~~~~
+``setOnSetupComplete``
+~~~~~~~~~~~~~~~~~~~~~~
+
+Specify a closure to be called once Envoy's engine finishes its async initialization/startup.
+
+When Envoy is instantiated, its initializer returns before all of its internal configuration
+completes. This interface provides the ability to observe when Envoy has completed its setup and is
+ready to start dispatching requests. Any requests sent through Envoy before this setup completes
+will be queued automatically, and this function is typically used purely for observability.
+
+**Example**::
+
+  // Kotlin
+  builder.setOnSetupComplete { /*do something*/ }
+
+  // Swift
+  builder.setOnSetupComplete { /*do something*/ }
```

Comment:
What about `onEngineRunning`?

There's a separate, potentially interesting callback point where objects have been initialized, but the main run loop hasn't started and configuration hasn't yet been loaded. `onEngineRunning` is maybe a little bit more explicit about the juncture to which it refers.

Reply:
Okay, updated.